tests/components/concurrency.test.ts

```diff
@@ -3696,6 +3696,42 @@ test("another scenario with delayed rendering", async () => {
   ]).toBeLogged();
 });
 
+test("delayed fiber does not get rendered if it was cancelled", async () => {
+  class D extends Component {
```

Comment:
add useLogLifeCycle on each component and check what is logged at every step. so this test will be more robust

Reply:
Cannot do that as it interferes with the timing heavily.